corehq/apps/export/tasks.py

```diff
@@ -208,3 +217,85 @@ def generate_schema_for_all_builds(self, schema_cls, domain, app_id, identifier)
         only_process_current_builds=False,
         task=self,
     )
+
+
+@periodic_task(run_every=crontab(hour="*", minute="30", day_of_week="*"),
+               queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def generate_incremental_exports():
+    incremental_exports = IncrementalExport.objects.filter(active=True)
+    for incremental_export in incremental_exports:
+        process_incremental_export.delay(incremental_export.id)
+
+
+@task
+def process_incremental_export(incremental_export_id):
+    incremental_export = IncrementalExport.objects.get(id=incremental_export_id)
+    checkpoint = _generate_incremental_export(incremental_export)
+    if checkpoint:
+        _send_incremental_export(incremental_export, checkpoint)
+
+
+def _generate_incremental_export(incremental_export):
+    export_instance = incremental_export.export_instance
+    export_instance.export_format = Format.UNZIPPED_CSV  # force to unzipped CSV
+    checkpoint = incremental_export.last_valid_checkpoint
+    filters = []
+    if checkpoint:
+        filters.append(ServerModifiedOnRangeFilter(gt=checkpoint.last_doc_date))
+
+    class LastDocTracker:
+        def __init__(self, doc_iterator):
+            self.doc_iterator = doc_iterator
+            self.last_doc = None
+            self.doc_count = 0
+
+        def __iter__(self):
+            for doc in self.doc_iterator:
+                self.last_doc = doc
+                self.doc_count += 1
+                yield doc
+
+    with TransientTempfile() as temp_path, metrics_track_errors('generate_incremental_exports'):
+        writer = get_export_writer([export_instance], temp_path, allow_pagination=False)
+        with writer.open([export_instance]):
+            query = _get_export_query(export_instance, filters)
+            # TODO: for some reason this sorting doesn't always seem to work in tests
+            query = query.sort('server_modified_on')  # reset sort to this instead of opened_on
+            docs = LastDocTracker(iter_es_docs_from_query(query))
+            write_export_instance(writer, export_instance, docs)
+
+        export_file = ExportFile(writer.path, writer.format)
+
+        if docs.doc_count <= 0:
+            return
+
+        new_checkpoint = incremental_export.checkpoint(
+            docs.doc_count, docs.last_doc.get('server_modified_on')
+        )
+
+        with export_file as file_:
+            # TODO: would be nice to save this compressed instead of uncompressed
+            # Could either be handled by the blobdb or else generated as a compressed CSV zip
+            db = get_blob_db()
+            db.put(
+                file_,
+                domain=incremental_export.domain,
+                parent_id=new_checkpoint.id,
+                type_code=CODES.incremental_data_export,
```

Comment:
@millerdev do you think it's reasonable to create a new code for this or just use `data_export` or `temp_file`

Reply:
I would use `data_export` (instead of creating a new type code) unless we know we want to do something like quantify the number and/or size of blobs stored for this particular type of export.

It's interesting that `IncrementalExportCheckpoint` uses `data_export`. Is that a different thing, or is that a mistake?

https://github.com/dimagi/commcare-hq/blob/698c9a5bf3ba20e2cf32957897bdbc0e2394b7ef/corehq/apps/export/models/incremental.py#L63

Reply:
That's a mistake, thanks. I'll update to use `data_export`